internal/MozziGuts.hpp

```diff
@@ -266,19 +269,51 @@ unsigned long mozziMicros() { return audioTicks() * MICROS_PER_AUDIO_TICK; }
 ////// BEGIN initialization ///////
 void startMozzi(int control_rate_hz) {
 #if !MOZZI_IS(MOZZI_ANALOG_READ, MOZZI_ANALOG_READ_NONE)
-  setupMozziADC(); // you can use setupFastAnalogRead() with FASTER_ADC or FASTEST_ADC
+  MozziPrivate::setupMozziADC(FAST_ADC); // you can use setupFastAnalogRead() with FASTER_ADC or FASTEST_ADC
                    // in setup() if desired (not for Teensy 3.* )
 #endif
   // delay(200); // so AutoRange doesn't read 0 to start with
   update_control_timeout = MOZZI_AUDIO_RATE / control_rate_hz;
   startAudio();
 }
 
+uint32_t MozziRandPrivate::x=132456789;
+uint32_t MozziRandPrivate::y=362436069;
+uint32_t MozziRandPrivate::z=521288629;
+
 ////// END initialization ///////
+}
 
 // reduce Macro leakage
 #undef LOOP_YIELD
 #undef BYPASS_MOZZI_OUTPUT_BUFFER
 #undef AUDIO_HOOK_HOOK
 #undef AUDIOTICK_ADJUSTMENT
 #undef MOZZI__LEGACY_AUDIO_INPUT_IMPL
+
+// "export" publicly accessible functions defined in this file
+// NOTE: unfortunately, we cannot just write using MozziPrivate::mozziMicros(), and that will conflict with, rather than define mozziMicros()
```

Comment:
I might be a bit slow, but for the sake of learning could you explain me why `MozziPrivate::startMozzi()` would not work for instance?

Reply:
In a vain attempt at keeping the source somewhat readable, I trying not to "enter" and "leave" the MozziPrivate-namespace too often. I.e. the basic idea was to write

```
namespace MozziPrivate {
  void privateFunction1() {}
  void privateFunction2() {}
  void publicFunction1() {}
  void privateFunction3() {}
}
void publicFuction1() { MozziPrivate::publicFunction1(); }
```

rather than:

```
namespace MozziPrivate {
  void privateFunction1() {}
}

namespace MozziPrivate {
  void privateFunction2() {}
}

void publicFunction1() {}

namespace MozziPrivate {
  void privateFunction3() {}
}
```

Maybe the second variant is preferable after all? I'm not sure at all (too bad we cannot just write ```void MozziPrivate::privateFunction1() { [definition] }```).

Reply:
Or perhaps this (ignoring convetional code-formatting rules)?

```namespace MozziPrivate { void privateFunction1() {
  [definition]
}}```

Reply:
But in that case, users would need to use the Mozzi namespace right? Or would the "export" part would still be there?

A global namespace (`mozzi::`) would be neat for all functions in my opinion but:

 - it might be intimidating to some users
 - this would break existing sketches badly (we are already there, but not *that* badly).

Reply:
Two somewhat separate but related topics, here:

As for the different variants of defining the MozziPrivate-namespace: The latter two variants would not need the "export" trick, because the "public" functions would not be placed inside MozziPrivate, in the first place. In contrast, the first variant (== current state of the PR) puts everything into MozziPrivate, and then "exports" the public ones.

I have indeed also been thinking a bit about adding a mozzi-namespace. Note that we could make it such that a default setup and/or examples could simply include a ```using namespace mozzi;``` line, which could again be turned off by some config switch, if desired. I have already added a MOZZI_COMPATIBILITY_LEVEL config define (mostly unused for now), which could also help with such a transition (while the compatibility level is set to < 2.0, the ```using``` line would be included by default, to keep old sketches working, but for >= 2.0, the default could be reversed).

In other words, the impact on users could be kept quite minimal, but without a doubt, we'd need to add a whole bunch of "namespace" statements, everywhere.